File: 0812-largest-triangle-area/0812-largest-triangle-area.py
class Solution:
    def largestTriangleArea(self, points: List[List[int]]) -> float:
        points.sort()
        ans, n = 0.0, len(points)
        height, width = defaultdict(list), defaultdict(list)

        for i in range(n):
            x, y = points[i]
            height[x].append(y)
            width[y].append(x)

        ans = 0
        for x, y_pts in height.items():
            if x in width:
                ans = max(ans, (width[x][-1] - width[x][0]) * (y_pts[-1] - y_pts[0]) / 2.0)

        return ans

File: 0812-largest-triangle-area/0812-largest-triangle-area.cpp
class Solution {
public:
    double largestTriangleArea(vector<vector<int>>& points) {
        int n = points.size();
        double ans = 0.0;

        for (int i = 0; i < n; ++i){
            pair<int, int> p = {points[i][0], points[i][1]};
            for (int j = i + 1; j < n; ++j){
                pair<int, int> q = {points[j][0], points[j][1]};
                for (int k = j + 1; k < n; ++k){
                    pair<int, int> r = {points[k][0], points[k][1]};
                    ans = max(ans, 0.5 * (abs((p.first * q.second + q.first * r.second + r.first * p.second) - (p.second * q.first + q.second * r.first + r.second * p.first))));
                }
            }
        }

        return ans;
    }
};